codemod/base.py

```diff
@@ -685,7 +686,18 @@ def _ask_about_patch(patch, editor, default_no):
         _save(patch.path, lines)
     if p in 'eE':
         run_editor(patch.start_position, editor)
+    if p in 'q':
+        sys.exit(0)
 
+def _getch():
+    fd = sys.stdin.fileno()
+    old_settings = termios.tcgetattr(fd)
+    try:
+        tty.setraw(sys.stdin.fileno())
+        ch = sys.stdin.read(1)
+    finally:
+        termios.tcsetattr(fd, termios.TCSADRAIN, old_settings)
```

Comment:
No, that is different. tty.setraw places the terminal in raw mode as opposed to canonical mode, which a terminal running a command-line interface is normally in. Canonical mode has the property that input is only sent to the application after <Enter> is pressed. As this is exactly what we want to avoid here, the terminal must be switched to raw mode momentarily. But it is vital that the terminal is returned to it's original state after catching the keystroke, and the try/finally does just that: The before state is saved in old_settings, and the finally clause ensures that it gets restored no matter what happens.